File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idx = {}
        for i in range(len(nums)):
            num = nums[i]
            requiredPrev = target - num
            if requiredPrev in idx:
                return [idx[requiredPrev], i]
            
            idx[num] = i

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> cache;
        auto numsIter = nums.begin();
        
        for (int i = 0; i < nums.size(); i++) {
            int other = target - *numsIter;
            if (cache.count(other)) {
                return {cache[other], i};
            }
            cache.insert({*numsIter, i});
            numsIter++;
        }
        
        return {-1, -1};
    }
};